src/prolog/machine/system_calls.rs

```diff
@@ -5137,8 +5139,27 @@ impl MachineState {
                 let result = Addr::HeapCell(self.heap.to_list(chars));
                 self.unify(version, result);
             }
+            &SystemClauseType::CryptoRandomByte => {
+                let arg = self[temp_v!(1)];
+                let mut bytes: Vec<u8> = vec![0];
```

Comment:
Given that you know the size at compile-time, I suggest you use a fixed-size array instead of a vector. Something like `let mut bytes = [0u8];` should work, you may need some tweaking for getting a slice :thinking:

Reply:
Thank you! Another approach could be to generate the N elements, and then use the Rust engine to convert the bytes to a list of integers internally, and return that. This would avoid round-trips between engine and Prolog layer. However, I would only recommend this if it can be done easily, because adding Rust code only for this specific use case is not worth it.